kubernetes/schema_pod_spec.go

```diff
@@ -4,66 +4,86 @@ import (
 	"github.com/hashicorp/terraform/helper/schema"
 )
 
-func podSpecFields(isUpdatable bool) map[string]*schema.Schema {
+func podSpecFields(isUpdatable, isDeprecated, isComputed bool) map[string]*schema.Schema {
+	var deprecatedMessage string
+	if isDeprecated {
+		deprecatedMessage = "This field is deprecated because template was incorrectly defined as a PodSpec preventing the definition of metadata. Please use the one under the spec field"
+	}
 	s := map[string]*schema.Schema{
 		"active_deadline_seconds": {
 			Type:         schema.TypeInt,
 			Optional:     true,
+			Computed:     isComputed,
 			ValidateFunc: validatePositiveInteger,
 			Description:  "Optional duration in seconds the pod may be active on the node relative to StartTime before the system will actively try to mark it failed and kill associated containers. Value must be a positive integer.",
+			Deprecated:   deprecatedMessage,
 		},
 		"container": {
 			Type:        schema.TypeList,
 			Optional:    true,
+			Computed:    isComputed,
 			Description: "List of containers belonging to the pod. Containers cannot currently be added or removed. There must be at least one container in a Pod. Cannot be updated. More info: http://kubernetes.io/docs/user-guide/containers",
+			Deprecated:  deprecatedMessage,
 			Elem: &schema.Resource{
 				Schema: containerFields(isUpdatable, false),
 			},
 		},
 		"init_container": {
 			Type:        schema.TypeList,
 			Optional:    true,
+			Computed:    isComputed,
 			ForceNew:    true,
 			Description: "List of init containers belonging to the pod. Init containers always run to completion and each must complete succesfully before the next is started. More info: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/",
+			Deprecated:  deprecatedMessage,
 			Elem: &schema.Resource{
 				Schema: containerFields(isUpdatable, true),
 			},
 		},
 		"dns_policy": {
-			Type:        schema.TypeString,
-			Optional:    true,
-			Default:     "ClusterFirst",
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: isComputed,
+			//Default:     "ClusterFirst"
```

Comment:
I suppose this is because of conflicts with `Computed` being set.
Let's not comment the `Default` out but remove it altogether.

Reply:
I've committed the change, let me know what you prefer.